examples/v2-minimal/packages/contracts/src/codegen/tables/EventTable.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+/* Autogenerated file. Do not edit manually. */
+
+// Import schema type
+import { SchemaType } from "@latticexyz/schema-type/src/solidity/SchemaType.sol";
+
+// Import store internals
+import { IStore } from "@latticexyz/store/src/IStore.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { StoreCore } from "@latticexyz/store/src/StoreCore.sol";
+import { Bytes } from "@latticexyz/store/src/Bytes.sol";
+import { Memory } from "@latticexyz/store/src/Memory.sol";
+import { SliceLib } from "@latticexyz/store/src/Slice.sol";
+import { EncodeArray } from "@latticexyz/store/src/tightcoder/EncodeArray.sol";
+import { Schema, SchemaLib } from "@latticexyz/store/src/Schema.sol";
+import { PackedCounter, PackedCounterLib } from "@latticexyz/store/src/PackedCounter.sol";
+
+bytes32 constant _tableId = bytes32(abi.encodePacked(bytes16(""), bytes16("event")));
+bytes32 constant EventTableTableId = _tableId;
+
+library EventTable {
+  /** Get the table's schema */
+  function getSchema() internal pure returns (Schema) {
+    SchemaType[] memory _schema = new SchemaType[](1);
+    _schema[0] = SchemaType.BOOL;
+
+    return SchemaLib.encode(_schema);
+  }
+
+  function getKeySchema() internal pure returns (Schema) {
+    SchemaType[] memory _schema = new SchemaType[](1);
+    _schema[0] = SchemaType.BYTES32;
+
+    return SchemaLib.encode(_schema);
+  }
+
+  /** Get the table's metadata */
+  function getMetadata() internal pure returns (string memory, string[] memory) {
+    string[] memory _fieldNames = new string[](1);
+    _fieldNames[0] = "value";
+    return ("EventTable", _fieldNames);
+  }
+
+  /** Register the table's schema */
+  function registerSchema() internal {
+    StoreSwitch.registerSchema(_tableId, getSchema(), getKeySchema());
+  }
+
+  /** Register the table's schema (using the specified store) */
+  function registerSchema(IStore _store) internal {
+    _store.registerSchema(_tableId, getSchema(), getKeySchema());
+  }
+
+  /** Set the table's metadata */
+  function setMetadata() internal {
+    (string memory _tableName, string[] memory _fieldNames) = getMetadata();
+    StoreSwitch.setMetadata(_tableId, _tableName, _fieldNames);
+  }
+
+  /** Set the table's metadata (using the specified store) */
+  function setMetadata(IStore _store) internal {
+    (string memory _tableName, string[] memory _fieldNames) = getMetadata();
+    _store.setMetadata(_tableId, _tableName, _fieldNames);
+  }
```

Comment:
wondering if `registerSchema` and `setMetadata` should also be ephemeral (only events) or not

feels weird to write these table definitions even though the underlying tables will never be written to

Reply:
that might be bad for sync, snap sync in particular. Also complicates tablegen a bit

Reply:
would tent to keeping this, since it might make it easier to sync the ephemeral events to off-chain indexers if this is ever wanted

Reply:
I guess `deleteRecord` is not really needed in this lib though (and a bit misleading)

Reply:
yeah, I missed deleteRecord because it has a separate rendering condition from the other record/field methods